### ISSUE TITLE
7-1 Learning rate, data preprocessing, overfitting

### ISSUE THREAD

**Issue by NahahaNahaha** [NahahaNahaha/tensorflow#14]
Machine learning 을 만들 때 주의해야할 점이 크게 3가지 있다. 
1. Learning rate 설정 
- 지금까지 수식상에서 알파로 쓴 learning rate 는 이상적인 값에 비해 많이 클 경우 overshooting 이라는 문제점을 발생시킨다. 
<img width="496" alt="2018-04-09 3 43 19" src="https://user-images.githubusercontent.com/26895961/38483114-c50f51d4-3c0c-11e8-8e82-7db84242ffa1.png"> 
이는 위와 같이 기울기에 너무 큰 값을 곱하다보니 cost 를 최소화 시키는 w의 값을 잘 찾지 못하고 이리저리 튕기게되는 것을 의미한다. 
반대로 너무 작을 경우는 도달하는데 너무 느리다는 단점이 있다. 
<img width="474" alt="2018-04-09 3 43 27" src="https://user-images.githubusercontent.com/26895961/38483158-efbe3364-3c0c-11e8-9d70-b8ce98e2cdf9.png">
이를 해결하기 위해서는 learning rate값을 변화시키면서 cost함수의 추이를 관찰한뒤 조정해야한다. 

2. Data processing 
- 주어지는 데이터값의 편차치가 너무 클 경우 데이터의 그래프가 발산되도록 그려지는데 이렇게 될 경우 minimum을 찾는 데 어려움을 겪는다. 
<img width="993" alt="2018-04-09 3 46 09" src="https://user-images.githubusercontent.com/26895961/38483253-40e66220-3c0d-11e8-88d8-8052d8bd5a7c.png">
위와 같이 찾아가는 과정에서 발산하는 걸 볼 수 있다. 
이를 해결하기 위한 방법이 두가지 소개되었는데 
1. zero-centered  
데이터의 중심을 0에 가깝게 하는 방법 
2. normalized data 
데이터들의 범위가 정규분포범위를 따르게 바꾸는것 
2번 같은 경우는 standarization 이라 해서 python code로도 지원된다. 

3. overfitting 
-overfitting 이란 우리가 만든 모델이 너무 test case에만 잘 적용되는 것을 뜻한다. 
이를 해결하려면 training data를 늘려 좀 더 일반적인 w의 값을 구하거나 
데이터가 중복된 것을 줄이거나 
Regularization을 하는 것이다. 
이른바 일반화는 weight이 큰 값을 가지지 않도록 하는 것이다. 
<img width="732" alt="2018-04-09 3 57 15" src="https://user-images.githubusercontent.com/26895961/38483645-b4472e06-3c0e-11e8-95ec-d67e2fc7ec31.png">
따라서 수식으로 쓰면 위와 같이 쓸 수 있는데 위에 W의 제곱의 합에 곱해진 상수의 값이 클수록 일반화를 많이 시킨다는 것이고 작을수록 쓰지 않는다는 이야기가 된다.
